error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/MongoDBRules.java

```diff
@@ -0,0 +1,31 @@
+package tech.picnic.errorprone.refasterrules;
+
+import static com.google.errorprone.refaster.ImportPolicy.STATIC_IMPORT_ALWAYS;
+import static com.mongodb.client.model.Filters.eq;
+
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import com.google.errorprone.refaster.annotation.UseImportPolicy;
+import com.mongodb.client.model.Filters;
+import org.bson.conversions.Bson;
+import tech.picnic.errorprone.refaster.annotation.OnlineDocumentation;
+
+/** Refaster rules related to MongoDB client expressions and statements. */
+@OnlineDocumentation
+final class MongoDBRules {
+  private MongoDBRules() {}
+
+  /** Avoid {@link Enum#toString()} invocations when invoking {@link Filters#eq(String, Object)}. */
+  static final class Eq {
+    @BeforeTemplate
+    Bson before(String field, Enum<?> value) {
+      return eq(field, value.toString());
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    Bson after(String field, Enum<?> value) {
+      return eq(field, value);
+    }
+  }
```

Comment:
This rule works if `Enum#toString` and the encoded value are equivalent.

This may not always be the case. Is there a way in Error Prone to identify e.g. whether `Enum#toString` is overwritten?

If we cannot guarantee it, is a `BugChecker` the better choice?

Reply:
We might be able to use a Matcher, we have a few examples here: https://github.com/PicnicSupermarket/error-prone-support/tree/master/refaster-support/src/main/java/tech/picnic/errorprone/refaster/matchers. 

I think we can infer something like you are describing in a Matcher 💪🏻 !